### PR TITLE
11271 Recipe Image upload

### PR DESCRIPTION
**Pull request by vicmaburrito**
## Use Rails "ActiveStorage" 
- [x] Display image on Recipe listing and view
- [x] Plan for the fact that someday business will want multiple pictures
- [x] Need to be signed in
- [x] Image displayed in gray box above "Steps" in its own section (no bigger than 100 x 100 px)
- [x] Image displayed on recipe listing on far right of gray box and fill up entire box with some padding (200 x 200 px)

## Getting Started
- Run ```rails db:migrate```

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -53,7 +53,7 @@ def set_recipe
     end
 
     def recipe_params
-      params.require(:recipe).permit(:name, :description, ingredient_ids: [])
+      params.require(:recipe).permit(:name, :description, :photo, ingredient_ids: [], images: [])
```
Comment (jm4rt1n3z) [inline app/controllers/recipes_controller.rb:56]: Wow, now it would be possible to add multiple images. good work

**Review by AlexRS90 — CHANGES_REQUESTED** [review]
## Hello Team,

### Good job so far!!

## Highlight:
✅ Good pull request description
✅ Descriptive commit messages
✅ Profesional PR

After change to your branch and follow the instructions in the description I got some errors, checkout the SS below 👇 

And without the migration if I run the server I got the following error:
image error

You are almost done!!

File: db/migrate/20220525183015_create_active_storage_tables.active_storage.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# This migration comes from active_storage (originally 20170806125915)
+class CreateActiveStorageTables < ActiveRecord::Migration[5.2]
+  def change
+    create_table :active_storage_blobs do |t|
```
Comment (coderdave) [inline db/migrate/20220525183015_create_active_storage_tables.active_storage.rb:4]: why are the tables called "active_storage"? Please use a different name for the tables. It should be something that makes sense for the app.

Comment (vicmaburrito) [reply db/migrate/20220525183015_create_active_storage_tables.active_storage.rb:4]: Hi There!, We were seeking about change active storage table name, but we realize about the command we ran to create the blob a active storage stuff with "rails active_storage:install" command creates a default name for the class and tables

**Review by cdelacruz8512 — APPROVED** [review]
Hey guys 

Good job, I tested the all image extensions (jpg, jpeg,png) and It works. I approve it.

**Review by jm4rt1n3z — APPROVED** [review]
hi guys, I checked the validation when a file size has more than 1 mb, it works fine. Good job! I approve it.

**Comment by vicmaburrito** [comment]
Hello! @coderdave, This issue it´s already solve [HERE](https://github.com/coderdave/myrecipes/pull/6/commits/6b645f764a58329630bc38d32c441557a5e0a012)👍🏾

**Review by faustino-garcia — APPROVED** [review]
Good job guys, I see you have made all the requested changes. I ran the test suite again and everything seems fine, also I follow your setup advice and I could make a functional test with different image extensions. 

All the validations worked as expected.

**Review by AlexRS90 — APPROVED** [review]
Hello team!!

Good job solving the issues, congratulations, there are nothing else to say other than is time to merge!!

File: app/models/recipe.rb [hunk 3]
```diff
@@ -8,5 +8,24 @@ class Recipe < ApplicationRecord
   validates :description, presence: true, length: { minimum: 5, maximum: 500 }
   validates :chef_id, presence: true
 
-  default_scope -> { order(updated_at: :desc)} 
+  default_scope -> { order(updated_at: :desc)}
+  has_many_attached :images, dependent: :destroy
```
Comment (coderdave) [inline app/models/recipe.rb:12]: remove

**Review by coderdave — APPROVED** [review]
good job

**Pull request merged**